core/src/test/java/org/apache/iceberg/TestIncrementalDataScan.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import java.util.List;
+import java.util.Set;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestIncrementalDataScan extends TableTestBase {
+
+  @Before
+  public void setupTableProperties() {
+    table.updateProperties().set(TableProperties.MANIFEST_MIN_MERGE_COUNT, "2").commit();
+  }
+
+  @Test
+  public void testInvalidScans() {
+    add(table.newAppend(), files("A"));
+    AssertHelpers.assertThrows(
+        "from and to snapshots cannot be the same, since from snapshot is exclusive and not part of the scan",
+        IllegalArgumentException.class, "from and to snapshots cannot be the same", () -> appendsBetweenScan(1, 1));
+  }
+
+  @Test
+  public void testAppends() {
+    add(table.newAppend(), files("A")); // 1
+    add(table.newAppend(), files("B"));
+    add(table.newAppend(), files("C"));
+    add(table.newAppend(), files("D"));
+    add(table.newAppend(), files("E")); // 5
+    Assert.assertEquals(Sets.newHashSet("B", "C", "D", "E"), appendsBetweenScan(1, 5));
+    Assert.assertEquals(Sets.newHashSet("C", "D", "E"), appendsBetweenScan(2, 5));
+  }
+
+  @Test
+  public void testReplaceOverwritesDeletes() {
```

Comment:
Nice job on these tests. They are easy to read and understand. Thank you!

Reply:
Thks!